lib/screens/product_edit_screen.dart

```diff
@@ -36,15 +42,16 @@ class _TBProductEditScreenState extends State<TBProductEditScreen> {
 
   @override
   void initState() {
-    _nameTextEditingController.text = widget.selectedProduct.name;
-    _companySelection = widget.selectedProduct.company;
-    _categorySelection = widget.selectedProduct.category;
-    _descriptionTextEditingController.text = widget.selectedProduct.description;
+    _nameTextEditingController.text =
+        ref.read(productProvider(widget.id)).value!.name;
+    _companySelection = ref.read(productProvider(widget.id)).value!.company;
+    _categorySelection = ref.read(productProvider(widget.id)).value!.category;
+    _descriptionTextEditingController.text =
+        ref.read(productProvider(widget.id)).value!.description;
     _discountTextEditingController.text =
-        widget.selectedProduct.discount.toStringAsFixed(2);
+        ref.read(productProvider(widget.id)).value!.discount.toStringAsFixed(2);
     _priceTextEditingController.text =
-        widget.selectedProduct.price.toStringAsFixed(2);
-
+        ref.read(productProvider(widget.id)).value!.price.toStringAsFixed(2);
     super.initState();
   }
 
```

Comment:
Instead of 
` if (_formKey.currentState != null) {
      _formKey.currentState!.dispose();
    }`
use 
`  _formKey.currentState?.dispose();`
it do the same